docstr_coverage/printers.py

```diff
@@ -0,0 +1,138 @@
+
+    Parameters
+    ----------
+    line: String
+        The text to print"""
+    logger.info(line)
+
+
+class LegacyPrinter:
+    """Printing functionality consistent with the original early-versions docstr-coverage outputs.
+
+    In future versions, the interface of this class will be refined and an abstract superclass
+    will be extracted. Thus, coding against the current interface will require refactorings with
+    future versions of docstr-coverage."""
+
+    def __init__(
+        self, verbosity, scip_magic, skip_file_docstring, skip_init, skip_class_def, skip_private,
+    ):
+        self.verbosity = verbosity
+        self.skip_magic = scip_magic
+        self.skip_file_docstring = skip_file_docstring
+        self.skip_init = skip_init
+        self.skip_class_def = skip_class_def
+        self.skip_private = skip_private
+
+    def print(self, results):
+        """Prints a provided set of results to stdout.
+
+        Parameters
+        ----------
+        results: ResultCollection
+            The information about docstr presence to be printed to stdout."""
+        if self.verbosity >= 2:
+            self._print_file_statistics(results)
+        if self.verbosity >= 1:
+            self._print_overall_statistics(results)
+
+    def _print_file_statistics(self, results):
+        """Prints the file specific information to stdout.
+
+        Parameters
+        ----------
+        results: ResultCollection
+            The information about docstr presence to be printed to stdout."""
+        for file_path, file in results.files():
+            # File Header
+            print_line('\nFile: "{}"'.format(file_path))
+
+            # List of missing docstrings
+            if self.verbosity >= 3:
+                if file.get_status() == FileStatus.EMPTY:
+                    print_line(" - File is empty")
+                for expected_docstr in file._expected_docstrings:
+                    if not expected_docstr.has_docstring and not expected_docstr.ignore_reason:
+                        if expected_docstr.node_identifier == "module docstring":
+                            print_line(" - No module docstring")
+                        else:
+                            print_line(
+                                " - No docstring for `{0}`".format(expected_docstr.node_identifier)
+                            )
+
+            # Statistics
+            count = file.count()
+            print_line(
+                " Needed: %s; Found: %s; Missing: %s; Coverage: %.1f%%"
+                % (count.needed, count.found, count.missing, count.coverage(),),
+            )
+            print_line()
+        print_line()
+
+    def _print_overall_statistics(self, results):
+        """Prints overall results (aggregated over all files) to stdout.
+
+        Parameters
+        ----------
+        results: ResultCollection
+            The information about docstr presence to be printed to stdout."""
+        if self.verbosity < 1:
```

Comment:
I think this check is unnecessary since `LegacyPrinter.print` already does it, and this method is only ever called by `print`. If for some crazy reason, someone called a non-public method like `_print_overall_statistics` directly, we might as well just print the stuff, assuming they know what they're doing